frontend/readhistory.lua

```diff
@@ -181,23 +182,38 @@ function ReadHistory:getFileByDirectory(directory, recursive)
 end
 
 --- Updates the history list after renaming/moving a file.
-function ReadHistory:updateItemByPath(old_path, new_path)
-    local index = self:getIndexByFile(old_path)
+function ReadHistory:updateItem(file, new_filepath)
+    local index = self:getIndexByFile(file)
     if index then
-        self.hist[index].file = new_path
-        self.hist[index].text = new_path:gsub(".*/", "")
+        local item = self.hist[index]
+        item.file = new_filepath
+        item.text = new_filepath:gsub(".*/", "")
+        self:_flush()
+    end
+end
+
+function ReadHistory:updateItems(files, new_path) -- files = { filepath = true, }
+    local history_updated
+    for file in pairs(files) do
+        local index = self:getIndexByFile(file)
+        if index then
+            local item = self.hist[index]
+            item.file = new_path .. "/" .. item.text
+            history_updated = true
+        end
+    end
+    if history_updated then
         self:_flush()
     end
 end
 
 --- Updates the history list after renaming/moving a folder.
 function ReadHistory:updateItemsByPath(old_path, new_path)
-    old_path = "^"..old_path
+    local l = #old_path
     local history_updated
     for i, v in ipairs(self.hist) do
-        local file, count = v.file:gsub(old_path, new_path)
-        if count == 1 then
-            self.hist[i].file = file
+        if v.file:sub(1, l) == old_path then
+            self.hist[i].file = new_path .. v.file:sub(l + 1)
```

Comment:
> `local l = #old_path`

The choice of `l` as the variable name makes this hard to read :) may be call it `len` ?